Add TransformComponent and Matrix

nattridgell opened the pull request:
Adds:
- TransformComponent
- Very basic TransformComponent tests
- Matrix (for transforms)
- Matrix.multiply functions for multiplying 2D and 3D matrices
  Note: Will be needed by TransformValue
- Matrix.is2DComponent function
- many tests for matrix multiplication using
  precomputed results

Commit:
Add TransformComponent

Adds the TransformComponent src and test files.
Note: asMatrix and is2DComponent methods should
throw errors as TransformComponent is an abstract type.

Commit:
Add Matrix for Transforms

Adds Matrix:
 - multiply functions for multiplying 2D and 3D matrices
   Note: Will be needed by TransformValue
 - cssString representation is matrix(...) or matrix3d(...)
 - is2DComponent if _matrix has length of 6
 - many tests for matrix multiplication using
   precomputed results

Commit:
Small style changes to Transforms

 - single line blocks need braces
 - remove Matrix from testing when already in scope
 - use internal.inherit instead of Object.create

wilddamon commented on src/matrix.js:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    this._matrix = [];
+    for (var index = 0; index < arguments.length; index++) {
```
It's more common just to use "i" as the variable name here.

nattridgell replied:
Cannot, i, j, and k are defined as parameters.

wilddamon commented on src/transform-component.js:
```diff
@@ -0,0 +1,35 @@
+  function TransformComponent() {
+  }
+
+  TransformComponent.prototype = {
```
I think we were moving to get rid of inline declarations like these... instead please use

TransformComponent.prototype.asMatrix = function() { ... }
TransformComponent.prototype.isDDComponent = function() { ... }

wilddamon commented on test/js/transform-component.js:
```diff
@@ -0,0 +1,13 @@
+suite('TransformComponent', function() {
+  test('TransformComponent is a TransformComponent', function() {
```
This seems redundant... no need to test language features?

Commit:
Style Updates to Transforms

- add missing ; to the end of = function () {};
- remove redundant test from TransformComponent
- remove inline declerations of TransfomrComponent.prototype

wilddamon commented:
LGTM
